### PR TITLE
Finish refactoring codebase.

### PR DESCRIPTION
**Pull request by movrsi**
driver.c now contains the following files.
- attach.c
- detach.c
- probe.c

While at it refactored the detach routine.

### CODE REVIEW

**Comment by Tatsh** [comment]
Let's keep the files separate. It makes things easier in editors when viewing things side-by-side. So please go back to C source and header pairs on for each function you moved.

**Comment by movrsi** [comment]
The header contained the definitions, it was not included in any of the c files. if you prefer it in separate c files that is fine, i will clean it up so it has a chance of compiling in the future.

**Comment by Tatsh** [comment]
Okay the driver.h was intended to be that way but the functions are still intended to be in separate files: `attach.c` etc. The reason is to make editing side-by-side easier. Anything that needs those functions will include `driver.h` and at link time they will be found.

**Comment by movrsi** [comment]
would it just be better to let them all have their own specific headers with their own specific constants or have a constants.h for all the constants to keep the codebase tidy?

**Comment by Tatsh** [comment]
There aren't a lot of constants so they can go in one file and be shared.

Also I do not understand the title of this PR.

**Comment by movrsi** [comment]
Well typically it's attach.h->attach.c. i have reverted and will commit the fixed up commits.

**Comment by Tatsh** [comment]
I don't see a need for the separate headers for each individual .c file. Linking will take care of finding the definitions. If this were a library you would declare all in a few headers and not necessarily correlating to the organisation of the source.

**Comment by movrsi** [comment]
There, rather than open a new pull just use this one. I finished refactoring the rest of the codebase.

**Comment by movrsi** [comment]
bumped since this has been sitting here for almost 2 months. not sure if this project is being focused on right now.

**Comment by movrsi** [comment]
@Tatsh Due to my account being suspended out of no where, it would be wise to review this pr and merge.

File: ps2-decompile/callbacks.c [hunk 1]
```diff
@@ -1,194 +1,220 @@
...
+            offset = i + 0x7804;
+
+#ifdef DEBUG
+           (void) printf("data at offset %i byte=%c", offset, g->data[offset]);
+#endif
+
             if (g->port) {
-                *(char *)(8 * (g->port - 1) + i + 0x2D20) = g->unk8[i + 0x7804];
+                // TODO figure out what the driver is doing here.
+                // it appears to be a memcpy.
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:33]: I do not think this is an accurate comment so please remove it.

What is going on below is accessing an array and assigning a value at some index. It also could be accessing a struct offset.

**Comment by Tatsh** [comment]
Run `clang-format -i ps2-decompile/*` and commit the changes.

File: ps2-decompile/callbacks.c [hunk 2]
```diff
@@ -1,194 +1,220 @@
...
+                *(char *)(8 * (g->port - 1) + i + 0x2D20) = g->data[offset];
                 kb[8 * g->port - 1] = 63;
             }
         }
-        // v4 - 32744 = 2CD4
-        printf((const char *)(v4 + 0xFFFF8018));
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:26]: Just comment out this printf. Do not delete it.

File: ps2-decompile/callbacks.c [hunk 3]
```diff
@@ -1,194 +1,220 @@
...
-    req.length = 0x400;
-    transferPipeRet =
-        sceUsbdTransferPipe(g->configEndpoint, report, 0x400, &req, get_report_done, g);
-    if (transferPipeRet)
-        // v4 - 32712 = 2CF4
-        transferPipeRet = printf("usbmouse%d: %s -> 0x%x\n",
-                                 g->port1,
-                                 (const char *)(gp0 + 0xFFFF8038),
-                                 transferPipeRet);
+        (void) printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdSetConfiguration", result);
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:180]: Why are the above the lines deleted?

File: ps2-decompile/callbacks.c [hunk 4]
```diff
@@ -1,194 +1,220 @@
...
         }
-        ++g->unk7;
-        req.requesttype = 0x80; // USB_DIR_IN
-        req.request = 6;        // USB_REQ_GET_DESCRIPTOR
-        req.value = LOWORD(g->unk7) | 0x300;
-        req.index = *(int *)&g->unk8[0x7800];
-        req.length = 2048;
-        setInterfaceRet = sceUsbdTransferPipe(
-            g->configEndpoint, &g->buf[0x2000 * g->unk7], 2048, &req, get_string_done, g);
+        v11 = (short *)((char *)v11 + 1);
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:151]: Why are the above lines deleted?

File: ps2-decompile/attach.c [hunk 5]
```diff
@@ -175,19 +160,19 @@ int usbmouse_attach(int devId) {
     if ((transferPipeRet = sceUsbdTransferPipe(g->configEndpoint, 0, 0,
                                                &deviceRequest, set_config_done, g))) {
 #ifdef DEBUG
-        printf("usbmouse%d: sceUsbdSetConfiguration -> 0x%x\n",
-               g->port1, transferPipeRet);
+        (void) printf("usbmouse%d: sceUsbdSetConfiguration -> 0x%x\n", g->port1, transferPipeRet);
 #endif
-        goto fail;
```
Comment (Tatsh) [inline ps2-decompile/attach.c:181]: Why is this removed?

File: ps2-decompile/callbacks.c [hunk 6]
```diff
@@ -1,194 +1,220 @@
...
     if (result)
-        printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdInterruptTransfer", result);
+        (void) printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdInterruptTransfer", result);
+#endif
+
     if (!result && g->descriptorChecksPassed) {
-        usbmouse_count = g->count + 1;
-        g->count = usbmouse_count;
-        printf("usbmouse%d: count=%d data=(", g->port1, usbmouse_count);
+        g->count += 1;
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:18]: Why are the above lines deleted? I do not want printf calls removed.

File: ps2-decompile/callbacks.c [hunk 7]
```diff
@@ -1,194 +1,220 @@
...
+
+#ifdef DEBUG
+        (void) printf("usbmouse%d: count=%d data=(", g->port1, g->count);
+#endif
+
         for (i = 0; i < 8; ++i) {
-            // v4 - 32752 = 2CCC
-            printf((const char *)(v4 + 0xFFFF8010), g->unk8[i + 0x7804]);
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:19]: Comment out the printf instead of deleting it.

File: ps2-decompile/callbacks.c [hunk 8]
```diff
@@ -1,194 +1,220 @@
...
-        g->configEndpoint,
-        g->buf,
-        0x800,
-        &req,
-        get_string_done,
-        g); // sceUsbdControlTransfer(g->configEndpoint, 0x80, USB_REQ_GET_DESCRIPTOR, (USB_DT_STRING << 8) | index, 0x800, ..., g->buf, get_string_done, g)
-    if (transferPipeRet)
-        transferPipeRet =
-            printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdSetInterface", transferPipeRet);
+
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:85]: Why are the above lines deleted?

File: ps2-decompile/callbacks.c [hunk 9]
```diff
@@ -1,194 +1,220 @@
...
-        for (j = 0; j < *v7 - 2; ++j) {
-            printf((const char *)(v4 + 0xFFFF8024), v7[j + 2]);
-            if (j % 16 == 15)
-                printf((const char *)(v4 + 0xFFFF8020));
-        }
-        printf((const char *)(v4 + 0xFFFF8020));
-        if (!g->unk7) {
-            *(int *)&g->unk8[0x7800] = v7[2] | (v7[3] << 8);
-            printf("lang_id = 0x%04x\n", *(unsigned short *)&g->unk8[0x7800]);
+
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:103]: Why are the above lines deleted?

File: ps2-decompile/callbacks.c [hunk 10]
```diff
@@ -1,194 +1,220 @@
...
-            v11 = (short *)*(&string_desc + g->unk7);
-        else
-            v11 = off_2C30;
-        for (k = 0; k < *v7 - 2; ++k) {
-            if (*(unsigned char *)v11 != v7[k + 2]) {
-                g->descriptorChecksPassed = 0;
-                printf("String Desc Error\n");
-            }
-            v11 = (short *)((char *)v11 + 1);
+
```
Comment (Tatsh) [inline ps2-decompile/callbacks.c:110]: Why are the above lines deleted?

File: ps2-decompile/callbacks.c (hunk 6 above)
Comment (movrsi) [reply ps2-decompile/callbacks.c:18]: Opps, forgot to put it within an ifdef. ill fix that.

File: ps2-decompile/callbacks.c (hunk 8 above)
Comment (movrsi) [reply ps2-decompile/callbacks.c:85]: line 80 it resides; the request was predefined in the header since its just static values.
